### DISCUSSION TITLE
Angaben zum Lieferdatum im Beispielprojekt

### DISCUSSION THREAD

**Discussion by dsshad** [LandrixSoftware/XRechnung-for-Delphi#17] [General]
Hallo,
ich habe mal eine ZUGFeRD-PDF-Datei testweise mit dem Validator der ZUGFeRD-Community prüfen lassen und da wurden mir diese Hinweise angezeigt - der KoSIT-Validator hatte nichts zu bemängeln:

> [VD-Valitool-117]-Die Angabe des Zeitpunkts der Lieferung oder sonstigen Leistung der Rechnung muss mit einem Format (z.B. "102" für YYYYMMDD) qualifiziert werden. Diese Angabe fehlt.
> /rsm:CrossIndustryInvoice/rsm:SupplyChainTradeTransaction/ram:ApplicableHeaderTradeDelivery/ram:ActualDeliverySupplyChainEvent/ram:OccurrenceDateTime/udt:DateTimeString/@format
> 
> [VD-Valitool-108]-Nach dem deutschen UStG §14 ist die Angabe des Zeitpunkts der Lieferung oder sonstigen Leistung eine Pflichtangabe. Die Angabe fehlt in der XML-Datei. Bitte prüfen Sie, ob Sie diese Angabe in Ihrer Geschäftsbeziehung machen müssen.
> /rsm:CrossIndustryInvoice/rsm:SupplyChainTradeTransaction/ram:ApplicableHeaderTradeDelivery/ram:ActualDeliverySupplyChainEvent/ram:OccurrenceDateTime/udt:DateTimeString
> 
> [VD-Valitool-71]-Es konnten keine Angaben wie z.B. Geschäftsführer, Handelsregistereintrag oder ähnliches gefunden werden. (SubjectCode=REG). Bitte prüfen Sie, ob Sie diese Daten ggf. im Freitext angegeben haben oder diese Angaben für Ihre Rechnungsstellung nicht relevant sind.

Beim letzten muss ich noch schauen was es damit auf sich hat, aber die beiden ersten hängen damit zusammen, dass die Angabe zu **.DeliveryInformation.ActualDeliveryDate** nur dann gemacht wird, wenn die Lieferanschrift angegeben wird. Ich hatte mich an das Beispielprojekt angelehnt, wo es so gemacht wird.

Aber ich glaube, dass der Hinweis, diese Gruppe sei nur dann zu verwenden, wenn die Lieferanschrift von der Erwerberanschrift abweicht, nur für die Gruppe "DELIVER TO ADDRESS" gilt (also .DeliveryInformation.Address.xxx) und nicht für die gesamte Gruppe "DELIVERY INFORMATION".

So lese ich zumindest die Angabe in der Spezifikation und wenn ich das Lieferdatum unabhängig von der Lieferanschrift angebe, hat auch der andere Validator nichts zu meckern.

Dachte das könnte dann im Beispielprojekt vielleicht angepasst werden.

**Comment by dsshad** [comment]
Wenn ich allerdings die oben erwähnte Änderung vornehme und das Lieferdatum unabhängig von der Lieferanschrift angebe, meckert der Validator bei einer im UBL-Format erzeugten XML-Datei über die leeren Angaben zum Ort der Lieferung, weil in der XML Datei in dem Fall dieser Block erzeugt wird:

```
  <cac:Delivery>
    <cbc:ActualDeliveryDate>2024-09-07</cbc:ActualDeliveryDate>
    <cac:DeliveryLocation>
      <cac:Address>
        <cbc:StreetName></cbc:StreetName>
        <cbc:CityName></cbc:CityName>
        <cbc:PostalZone></cbc:PostalZone>
        <cac:Country>
          <cbc:IdentificationCode></cbc:IdentificationCode>
        </cac:Country>
      </cac:Address>
    </cac:DeliveryLocation>
  </cac:Delivery>

```
Angaben zur Lieferanschrift (BG-15) sollen ja nur gemacht werden, wenn sich diese von der Erwerberanschrift unterscheidet (in meinem Fall sind beide gleich).

Habe ich einen Denkfehler bei all dem oder ist die Erzeugung der leeren "Felder" bei der UBL-Ausgabe nicht richtig?

**Comment by landrix** [comment]
Das Problem sollte jetzt behoben sein.
